nilearn/_utils/niimg_conversions.py

```diff
@@ -352,9 +352,6 @@ def check_niimg(
     # Otherwise, it should be a filename or a SpatialImage, we load it
     niimg = load_niimg(niimg, dtype=dtype)
 
-    if safe_get_data(niimg).size == 0:
-        raise ValueError("The image is empty.")
-
```

Comment:
I was wondering if instead of putting this check in `check_niimg`, creating a function `is_img_empty` and doing the check only where it is necessary would help with performance? Because `check_niimg` is called everywhere. If I am not mistaken, we have to check this only when `fit` is called, where it already needs to get image data.